data/data-pipeline/data_pipeline/tests/score/test_calculation.py

```diff
@@ -0,0 +1,288 @@
+# flake8: noqa: W0613,W0611,F811
+from dataclasses import dataclass
+import data_pipeline.score.field_names as field_names
+from data_pipeline.utils import get_module_logger
+from data_pipeline.score.score_narwhal import ScoreNarwhal
+from .fixtures import final_score_df  # pylint: disable=unused-import
+
+logger = get_module_logger(__name__)
+
+
+@dataclass
+class PercentileTestConfig:
+    percentile_column_name: str
+    threshold_column_name: str
+    threshold: float
+    percentile_column_need_suffix: bool = True
+
+    @property
+    def full_percentile_column_name(self):
+        if self.percentile_column_need_suffix:
+            return (
+                self.percentile_column_name
+                + field_names.PERCENTILE_FIELD_SUFFIX
+            )
+        return self.percentile_column_name
+
+
+### TODO: we need to blow this out for all eight categories
+def _check_percentile_against_threshold(df, config: PercentileTestConfig):
+    """Note - for the purpose of testing, this fills with False"""
+    is_minimum_flagged_ok = (
+        df[df[config.threshold_column_name].fillna(False)][
+            config.full_percentile_column_name
+        ].min()
+        >= config.threshold
+    )
+
+    is_maximum_not_flagged_ok = (
+        df[~df[config.threshold_column_name].fillna(False)][
+            config.full_percentile_column_name
+        ].max()
+        < config.threshold
+    )
+    errors = []
+    if not is_minimum_flagged_ok:
+        errors.append(
+            f"For column {config.threshold_column_name}, there is someone flagged below {config.threshold} percentile!"
+        )
+    if not is_maximum_not_flagged_ok:
+        errors.append(
+            f"For column {config.threshold_column_name}, there is someone not flagged above {config.threshold} percentile!"
+        )
+    return errors
+
+
+def test_percentile_columns(final_score_df):
+    low_income = PercentileTestConfig(
```

Comment:
So having done this, I started wanting to make it fancier --- like make the list happen automatically via some metaprogramming, and be able to impute the names of fields that follow a convention. But also I wanted to keep the test as obvious as possible, so I stuck with what I had. Honestly writing the test reinforced that the actual score code itself could use some refactoring to reduce duplication there, and then the test could rely on stronger conventions within tho score code to be further simplified.